src/usr/hdat/hdatiplparms.C

```diff
@@ -724,18 +725,97 @@ void HdatIplParms::hdatGetSystemParamters()
 
     this->iv_hdatIPLParams->iv_sysParms.hdatSplitCoreMode = 1;
 
-    TARGETING::ATTR_SYSTEM_BRAND_NAME_type l_systemBrandName = {0};
-    if(l_pSysTarget->tryGetAttr<TARGETING::ATTR_SYSTEM_BRAND_NAME>
-                                                         (l_systemBrandName))
+    /* Read UTIL::F5 keyword (maxlen=16)*/
+    errlHndl_t l_err = NULL;
+    size_t  dataSize = 0;
+    char    dataBuffer[17]={0};
+    char    sysVendorname[64]={0};
+    Target* node_tgt = UTIL::getCurrentNodeTarget();
+
+    l_err = deviceRead(node_tgt, NULL, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F5 ));
+
+    if (l_err || (dataSize > sizeof(dataBuffer) - 1))
     {
-        strcpy(reinterpret_cast<char*>
-                   (this->iv_hdatIPLParams->iv_sysParms.hdatSystemVendorName),
-                                                           l_systemBrandName);
+        HDAT_ERR("Read UTIL::F5 length error, len=%d...", dataSize);
+    }
+    else
+    {
+        memset(dataBuffer, 0, sizeof(dataBuffer));    
+        l_err = deviceRead(node_tgt, dataBuffer, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F5 ));
+        if (l_err)
+        {
+            HDAT_ERR("Read UTIL::F5 data error");
+        }
+        else if(strlen(dataBuffer) > sizeof(sysVendorname) - 1)
+        {
+            HDAT_ERR("Read UTIL::F5 string len override size: %d", strlen(dataBuffer));
+        }
+        else
+        {
+            HDAT_DBG("Read UTIL::F5 success len=%d str=%s", dataSize, dataBuffer);
+            if (strlen(dataBuffer))
+            {
+                strcat(sysVendorname, dataBuffer);
+            }
+        }
+    }
+
+    /* Read UTIL::F6 keyword(maxlen=16) */
+    dataSize = 0;
+    l_err = deviceRead(node_tgt, NULL, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F6 ));
+
+    if (l_err || (dataSize > sizeof(dataBuffer) - 1))
+    {
+        HDAT_ERR("Read UTIL::F6 length error...");
+    }
+    else
+    {
+        memset(dataBuffer, 0, sizeof(dataBuffer));
+        l_err = deviceRead(node_tgt, dataBuffer, dataSize, DEVICE_PVPD_ADDRESS( PVPD::UTIL, PVPD::F6 ));
+        if (l_err)
+        {
+            HDAT_ERR("Read UTIL::F6 data error");
+        }
+        else if(strlen(dataBuffer) + strlen(sysVendorname) > sizeof(sysVendorname) - 1)
+        {
+            HDAT_ERR("Read UTIL::F5+F6 string len override size: %d", strlen(dataBuffer) + strlen(sysVendorname));
+        }
+        else
+        {
+            HDAT_DBG("Read UTIL::F6 success len=%d str=%s", dataSize, dataBuffer);
+            if (strlen(dataBuffer))
+            {
+                if (strlen(sysVendorname))
+                {
+                    strcat(sysVendorname, " ");            // Note that we need a space here
```

Comment:
Why is a space needed?  I thought that F5+F6 was just considered to be a single string that happened to be split in half.  It seems like there is some extra interpretation of the exact contents that we don't really want.  (For example, a brand that had a single word that was more than 16 bytes in length wouldn't come out right.)  If a space is required it should be part of the contents of the keyword(s) themselves, not applied by this code.

Reply:
I got an error message from the email I received.
The email says that F5 is "Inpur" and F6 is "power system".
I understand that F5 is the vendor and F6 is something else.
Now, the space has been removed.

Reply:
Looking back at the email I can definitely see why you did what you did.  Looks like it was just some imprecision in the example.  The space will need to be included in the string passed into vpd-tool.